Makefile

```diff
@@ -10,12 +10,20 @@ cidocker = docker run ${ciargs} ${dockerargs} golang:1.4
 all: ${build_image}.created build check integration-tests
 
 integration-tests:
+	@if [ ! -e ./cmd/dnet/dnet ]; then \
+		echo "Dnet binary not present, triggering build"; \
+		make build; \
+	fi
 	@if [ ! -d ./integration-tmp ]; then \
 	    mkdir -p ./integration-tmp;	\
 	    git clone https://github.com/sstephenson/bats.git ./integration-tmp/bats; \
 	    ./integration-tmp/bats/install.sh ./integration-tmp; \
 	fi
 	@./integration-tmp/bin/bats ./test/integration/dnet
+	@if [ -e ./cmd/dnet/dnet ]; then \
```

Comment:
And I don't think we need to remove this. If any dependency of dnet changes (we can include dependencies in the above dnet target) then it will be rebuilt automatically. Otherwise there is no harm in keeping the old one there so that we don't need to trigger dnet build all the time we run `make integration-tests`